core/domain/html_validation_service.py

```diff
@@ -909,6 +909,46 @@ def get_invalid_svg_tags_and_attrs(svg_string):
     return (invalid_elements, invalid_attrs)
 
 
+def validate_math_tags_in_html(html_string):
+    """Returns a list of all invalid math tags in the given HTML.
+
+    Args:
+        html_string: str. The HTML string.
+
+    Returns:
+        list(dict). A list of dicts having invalid math tags. Each dict has
+        2 keys, one key is for the invalid tag and the other key is the error
+        raised because of the invalid tag.
+    """
+
+    soup = bs4.BeautifulSoup(
+        html_string.encode(encoding='utf-8'), 'html.parser')
+    error_list = []
+    for math_tag in soup.findAll(name='oppia-noninteractive-math'):
+        if math_tag.has_attr('raw_latex-with-value'):
+            try:
+                # The raw_latex attribute value should be enclosed in
+                # double quotes(&amp;quot;) and should be a valid unicode
+                # string.
+                raw_latex = (
+                    json.loads(unescape_html(math_tag['raw_latex-with-value'])))
+                objects.UnicodeString.normalize(raw_latex)
+            except Exception as e:
+                error_message = (
+                    'Invalid raw_latex value found in the math tag : %s' % (
+                        python_utils.UNICODE(e)))
+                error_list.append({
```

Comment:
You'll need to make a custom domain object if you want to do this. It may be simpler to just have a string representation of the math tag be part of the error message.

Reply:
added only the invalid tag as the part of error message.